src/Http/Http.Abstractions/test/UsePathBaseExtensionsTests.cs

```diff
@@ -238,6 +237,27 @@ private static HttpContext CreateRequest(string pathBase, string requestPath)
 
     private static ApplicationBuilder CreateBuilder()
     {
-        return new ApplicationBuilder(serviceProvider: null!);
+        return new ApplicationBuilder(new DummyServiceProvider());
+    }
+
+    private class DummyServiceProvider : IServiceProvider
+    {
+        private readonly Dictionary<Type, object> _services = new Dictionary<Type, object>();
+
+        public void AddService(Type type, object value) => _services[type] = value;
+
+        public object? GetService(Type serviceType)
+        {
+            if (serviceType == typeof(IServiceProvider))
+            {
+                return this;
+            }
+
+            if (_services.TryGetValue(serviceType, out var value))
+            {
+                return value;
+            }
+            return null;
```

Comment:
```suggestion
            }
            
            return null;
```

Reply:
Actually, I'm pretty happy with the nit! I kept the DummyServiceProvider the same as in the UseMiddlewareTests class for consistency purposes, but disliked the format. Would you like me to make the same edit in the UseMiddlewareTests class?